lib/portlayer/exec/container.go

```diff
@@ -340,12 +339,27 @@ func (c *Container) start(ctx context.Context) error {
 
 		// TODO: mechanism to trigger reinspection of long term transitional states
 		finalState = StateStarting
+		c.updateState(finalState)
+
 		return err
 	}
 
-	finalState = StateRunning
+	c.m.Lock()
+	defer c.m.Unlock()
 
-	return err
+	// Obtain the current state and set the final state to Running only if it's Starting.
+	// The current state is already Stopped if the container's process has exited or
+	// a poweredoff event has been processed.
+	currentState := c.state
+	if currentState == StateStarting {
+		finalState = StateRunning
+	} else {
+		log.Infof("current state of container %s is %v", c.ExecConfig.ID, currentState)
```

Comment:
I put this in because it would be helpful to log the current state since we skip a state change to `Running` here. You're right though, since we'd have already logged the existing state earlier. I'll remove this log.